platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/service/WritablePlatformStateStore.java

```diff
@@ -0,0 +1,565 @@
+                .legacyRunningEventHash(
+                        legacyRunningEventHash == null ? Bytes.EMPTY : legacyRunningEventHash.getBytes()));
+    }
+
+    @Override
+    public void setConsensusTimestamp(@NonNull final Instant consensusTimestamp) {
+        requireNonNull(consensusTimestamp);
+        final var previousState = stateOrThrow();
+        update(previousState
+                .copyBuilder()
+                .consensusSnapshot(previousState
+                        .consensusSnapshotOrElse(ConsensusSnapshot.DEFAULT)
+                        .copyBuilder()
+                        .consensusTimestamp(toPbjTimestamp(consensusTimestamp))));
+    }
+
+    @Override
+    public void setRoundsNonAncient(final int roundsNonAncient) {
+        final var previousState = stateOrThrow();
+        update(previousState.copyBuilder().roundsNonAncient(roundsNonAncient));
+    }
+
+    @Override
+    public void setSnapshot(@NonNull final com.swirlds.platform.consensus.ConsensusSnapshot snapshot) {
+        requireNonNull(snapshot);
+        final var previousState = stateOrThrow();
+        update(previousState.copyBuilder().consensusSnapshot(toPbjConsensusSnapshot(snapshot)));
+    }
+
+    @Override
+    public void setFreezeTime(@Nullable final Instant freezeTime) {
+        final var previousState = stateOrThrow();
+        update(previousState.copyBuilder().freezeTime(toPbjTimestamp(freezeTime)));
+    }
+
+    @Override
+    public void setLastFrozenTime(@Nullable final Instant lastFrozenTime) {
+        final var previousState = stateOrThrow();
+        update(previousState.copyBuilder().lastFrozenTime(toPbjTimestamp(lastFrozenTime)));
+    }
+
+    @Override
+    public void setFirstVersionInBirthRoundMode(@NonNull final SoftwareVersion firstVersionInBirthRoundMode) {
+        requireNonNull(firstVersionInBirthRoundMode);
+        final var previousState = stateOrThrow();
+        update(previousState
+                .copyBuilder()
+                .firstVersionInBirthRoundMode(firstVersionInBirthRoundMode.getPbjSemanticVersion())
+                .build());
+    }
+
+    @Override
+    public void setLastRoundBeforeBirthRoundMode(final long lastRoundBeforeBirthRoundMode) {
+        final var previousState = stateOrThrow();
+        update(previousState
+                .copyBuilder()
+                .lastRoundBeforeBirthRoundMode(lastRoundBeforeBirthRoundMode)
+                .build());
+    }
+
+    @Override
+    public void setLowestJudgeGenerationBeforeBirthRoundMode(final long lowestJudgeGenerationBeforeBirthRoundMode) {
+        final var previousState = stateOrThrow();
+        update(previousState
+                .copyBuilder()
+                .lowestJudgeGenerationBeforeBirthRoundMode(lowestJudgeGenerationBeforeBirthRoundMode));
+    }
+
+    @Override
+    public void bulkUpdate(Consumer<PlatformStateAccessor> updater) {
+        final var accumulator = new PlaformStateValueAccumulator();
```

Comment:
param `updater` missing nullity annotation and final

Reply:
Fixed